src/query/shorthand.ts

```diff
@@ -403,23 +403,32 @@ export namespace shorthandParser {
   }
 
   export function fn(encQ: EncodingQuery, fieldDefShorthand: string): EncodingQuery {
-    let func = fieldDefShorthand.substring(0, fieldDefShorthand.indexOf('('));
-    let insideFn = fieldDefShorthand.substring(func.length + 1, fieldDefShorthand.length - 1);
-    let insideFnParts = splitWithTail(insideFn, ',', 2);
-
-    // TODO(https://github.com/uwdata/compassql/issues/261)
-    // Parser should support enumspec values
-    if (func === '?') {
-      encQ.aggregate = '?';
-      encQ.bin = '?';
-      encQ.timeUnit = '?';
-      return rawFieldDef(encQ, insideFnParts);
-    } else if (AGGREGATE_OP_INDEX[func]) {
-      return aggregate(encQ, func, insideFnParts);
-    } else if (MULTI_TIMEUNIT_INDEX[func] || SINGLE_TIMEUNIT_INDEX[func]) {
-      return timeUnit(encQ, func, insideFnParts);
-    } else if (func === 'bin') {
-      return bin(encQ, func, insideFnParts);
+    // Aggregate, Bin, TimeUnit as enum spec case
+    if (fieldDefShorthand[0] === '?') {
+      let closingBraceIndex = getClosingBraceIndex(1, fieldDefShorthand);
+      let enumFnString = fieldDefShorthand.substring(2, closingBraceIndex);
```

Comment:
This is definitely incorrect for
```
{"aggregate": ["max", "min"], "bin": [false,true]}
```

Why don't you just parse the JSON and check if it's '?' or array

If it's array, assign `{enum:  <array>}` otherwise, `SHORT_ENUM_SPEC`.